finetune/target_models/deployment_model.py

```diff
@@ -0,0 +1,340 @@
+            else:
+                self.pipeline = self.pipe_gen()(self.config)
+        return self.pipeline.text_to_tokens_mask(X)
+
+    def get_target_input_fn(self, features, batch_size=None):
+        batch_size = batch_size or self.config.batch_size
+        features = pd.DataFrame(features).to_dict('list')
+        for key in features:
+            features[key] = np.array(features[key])
+        tf_dataset = lambda : tf.data.Dataset.from_tensor_slices(dict(features)).batch(batch_size)
+        return tf_dataset
+
+
+class DeploymentModel(BaseModel):
+    """ 
+    Implements inference in arbitrary tasks in a cached manner by loading weights efficiently, allowing for quick interchanging of
+    weights while avoiding slow graph recompilation.
+
+    :param config: A :py:class:`finetune.config.Settings` object or None (for default config).
+    :param \**kwargs: key-value pairs of config items to override.
+    """
+    def __init__(self, base_model, **kwargs):
+        """
+        For a full list of configuration options, see `finetune.config`.
+
+        :param base_model: One of the base models from finetune.base_models, excluding textcnn.
+        :param **kwargs: key-value pairs of config items to override.
+        """
+        if base_model not in [GPTModel, GPTModelSmall, BERTModelCased, GPT2Model]:
+            raise FinetuneError("Selected base model not supported.")
+        self.config = get_config(**kwargs)
+        self.validate_config()
+        self.input_pipeline = DeploymentPipeline(self.config)
+        super().__init__(**kwargs)
+        self.config.base_model = base_model
+        self.task = TaskMode.CLASSIFICATION
+        self.input_pipeline.task = self.task
+        self.featurizer_loaded = False
+        self.adapters = False
+
+    def load_featurizer(self):
+        """
+        Performs graph compilation of the featurizer, saving most compilation overhead from occurring at predict time. Should
+        be called after initialization but BEFORE any calls to load_trainables or predict.
+        """
+        self.featurizer_est = self._get_estimator('featurizer')
+        self.predict_hooks.feat_hook.model_portion = 'whole_featurizer'
+        for hook in self.predict_hooks:
+            hook.need_to_refresh = True
+        output = self.predict(['finetune'], exclude_target=True) #run arbitrary predict call to compile featurizer graph
+        self.featurizer_loaded = True
+
+    def load_trainables(self, path):
+        """
+        Load in a model from save file. Must be called after 
+        """
+        if not self.featurizer_loaded:
+            raise FinetuneError('Need to call load_featurizer before loading weights from file.')
+        original_model = self.saver.load(path)
+        if original_model.config.adapter_size is None:
+            LOGGER.warning("Loading without adapters will result in slightly slower prediction than models that use adapters.")
```

Comment:
Kind of a misleading warning if I understand correctly.  It's not that prediction will be slower, it's that loading will take longer and that additional cost will be incurred the next time an adapter file is loaded.